src/sidebar/app/components/Footer.js

```diff
@@ -0,0 +1,333 @@
+import React from 'react';
+
+import {getPadStats} from '../editor.js';
+
+/**
+ * Formats time for the Notes footer
+ * @param time
+ * @returns {string}
+ */
+function formatFooterTime(date) {
+  date = date || Date.now();
+  return new Date(date).toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'});
+}
+
+/**
+ * Set animation on footerButtons toolbar
+ *
+ *
+ * @param {Boolean} animateSyncIcon Start looping animation on sync icon
+ * @param {Boolean} syncingLayout   if true, animate to syncingLayout (sync icon on right)
+ *                                  if false, animate to savingLayout (sync icon on left)
+ * @param {Boolean} warning         Apply yellow warning styling on toolbar
+ */
+function setAnimation( animateSyncIcon = true, syncingLayout, warning ) { // animateSyncIcon, syncingLayout, warning
+  const footerButtons = document.getElementById('footer-buttons');
+  const enableSync = document.getElementById('enable-sync');
+  const savingIndicator = document.getElementById('saving-indicator');
+
+  if (animateSyncIcon === true && !footerButtons.classList.contains('animateSyncIcon')) {
+    footerButtons.classList.add('animateSyncIcon');
+  } else if (animateSyncIcon === false && footerButtons.classList.contains('animateSyncIcon')) {
+    footerButtons.classList.remove('animateSyncIcon');
+  }
+
+  if (syncingLayout === true && footerButtons.classList.contains('savingLayout')) {
+    footerButtons.classList.replace('savingLayout', 'syncingLayout');
+    enableSync.style.backgroundColor = 'transparent';
+    // Start blink animation on saving-indicator
+    savingIndicator.classList.add('blink');
+    // Reset CSS animation by removeing class
+    setTimeout(() => savingIndicator.classList.remove('blink'), 400);
+  } else if (syncingLayout === false && footerButtons.classList.contains('syncingLayout')) {
+    // Animate savingIndicator text
+    savingIndicator.classList.add('blink');
+    setTimeout(() => savingIndicator.classList.remove('blink'), 400);
+    setTimeout(() => {
+      enableSync.style.backgroundColor = null;
+    }, 400);
+    //
+    footerButtons.classList.replace('syncingLayout', 'savingLayout');
+  }
+
+  if (warning === true && !footerButtons.classList.contains('warning')) {
+    footerButtons.classList.add('warning');
+  } else if (warning === false && footerButtons.classList.contains('warning')) {
+    footerButtons.classList.remove('warning');
+  }
+}
+
+
+const SURVEY_PATH = 'https://qsurvey.mozilla.com/s3/notes?ref=sidebar';
+
+
+class Footer extends React.Component {
```

Comment:
The functions above for this component make this component file really long and complex, can we move the utils above out of here?

Reply:
Agree, refactoring is coming.